### ISSUE TITLE
prefer-global allow dynamic

### ISSUE THREAD

**Issue by remcohaszing** [mysticatea/eslint-plugin-node#238]
I like the `prefer-global` rule, but sometimes a dynamic require is needed inside a function. I suggest to add a boolean option `allowDynamic`. If this is set to true, this rule only reports nested imports if a literal string or template string is passed.

For example:

valid:

```js
require('./bar');

function foo(bar) {
  return require('./' + bar);
}

function foo(name) {
  return require(`./${bar}`);
}
```

invalid:

```js
function foo(name) {
  return require('./bar');
}

function foo(name) {
  return require(`./bar`);
}
```
